src/util/usePreLoader.ts

```diff
@@ -0,0 +1,50 @@
+import {Story} from "../interfaces";
+import {useEffect, useRef, useState} from "react";
+
+
+// Caches given Story[] using HTMLImageElement and HTMLVideoElement
+const cacheContent = async (contents: Story[]) => {
+	const promises = contents.map((content) => {
+		return new Promise(function (resolve, reject) {
+			if(!content.url) return
+
+			if(content.type === 'video') {
+				const video = document.createElement('video');
+				video.src = content.url;
+				video.onloadeddata = resolve;
+				video.onerror = reject;
+				return;
+			}
+
+			const img = new Image();
+			img.src = content.url;
+			img.onload = resolve;
+			img.onerror = reject;
+		})
+	})
+
+	await Promise.all(promises);
+}
+
+
+// Preloads images and videos from given Story[] using a cursor and preloadCount
+// Preload count is the number of images/videos to preload after the cursor
+// Cursor is the current index to start preloading from
+export const usePreLoader = (contents: Story[], cursor: number, preloadCount: number) => {
+	const urlsLoaded = useRef([] as string[]).current;
```

Comment:
A small nitpick, but `urlsLoaded` could be a `Set` since we only want to check existence of already loaded URLs.

Reply:
Sure, so a global Set?

Reply:
Btw. Will try to fix the last things this week :)